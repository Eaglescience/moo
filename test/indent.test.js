
const fs = require('fs')

const moo = require('../moo')
const indented = require('../indent')
const compile = moo.compile

describe('indent', () => {

  test("example", () => {

    const lexer = moo.compile({
      ws: /[ \t]+/,
      nl: { match: /(?:\r\n?|\n)+/, lineBreaks: true },
      id: /\w+/,
    })

    const tokens = indented(lexer, `
    if this
      if that
        another
    else
      there
    `)

    const output = Array.from(tokens)
    //for (const tok of output) console.log(tok)
    expect(output).toMatchSnapshot()

  })

})
