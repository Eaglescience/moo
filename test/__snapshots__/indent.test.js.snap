// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indent example 1`] = `
Array [
  Object {
    "col": 5,
    "line": 2,
    "lineBreaks": 0,
    "offset": 5,
    "toString": [Function],
    "type": "id",
    "value": "if",
  },
  Object {
    "col": 8,
    "line": 2,
    "lineBreaks": 0,
    "offset": 8,
    "toString": [Function],
    "type": "id",
    "value": "this",
  },
  Object {
    "type": "indent",
  },
  Object {
    "col": 7,
    "line": 3,
    "lineBreaks": 0,
    "offset": 19,
    "toString": [Function],
    "type": "id",
    "value": "if",
  },
  Object {
    "col": 10,
    "line": 3,
    "lineBreaks": 0,
    "offset": 22,
    "toString": [Function],
    "type": "id",
    "value": "that",
  },
  Object {
    "type": "indent",
  },
  Object {
    "col": 9,
    "line": 4,
    "lineBreaks": 0,
    "offset": 35,
    "toString": [Function],
    "type": "id",
    "value": "another",
  },
  Object {
    "type": "dedent",
  },
  Object {
    "type": "dedent",
  },
  Object {
    "col": 5,
    "line": 5,
    "lineBreaks": 0,
    "offset": 47,
    "toString": [Function],
    "type": "id",
    "value": "else",
  },
  Object {
    "type": "indent",
  },
  Object {
    "col": 7,
    "line": 6,
    "lineBreaks": 0,
    "offset": 58,
    "toString": [Function],
    "type": "id",
    "value": "there",
  },
  Object {
    "type": "dedent",
  },
]
`;
